### PR TITLE
[KOGITO-4952] Support installing product operator using insecure regi…

### PR DESCRIPTION
**Pull request by sutaakar**
…stry

Signed-off-by: Karel Suta <ksuta@redhat.com>

https://issues.redhat.com/browse/KOGITO-4952

Depends on https://github.com/kiegroup/kogito-operator/pull/848

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

**Comment by sutaakar** [comment]
Currently waiting for https://github.com/kiegroup/rhpam-kogito-operator/pull/12 to be merged first

**Review by radtriste — APPROVED** [review]

File: test/pkg/installers/rhpam_kogito_installer.go [hunk 1]
```diff
@@ -89,13 +90,28 @@ func GetRhpamKogitoInstaller() (installers.ServiceInstaller, error) {
...
 	}
 
-	yamlContent = strings.ReplaceAll(yamlContent, "quay.io/kiegroup/rhpam-kogito-operator:"+version.Version, framework.GetOperatorImageNameAndTag())
+	// Use insecure ImageStream when deploying on OpenShift to support using insecure registries
+	if framework.IsOpenshift() {
+		if err := framework.CreateInsecureImageStream(rhpamKogitoNamespace, rhpamKogitoImageStreamName, config.GetOperatorImageTag(), framework.GetOperatorImageNameAndTag()); err != nil {
+			return err
+		}
+
+		rhpamKogitoInternalImageTagName := fmt.Sprintf("image-registry.openshift-image-registry.svc:5000/%s/%s:%s", rhpamKogitoNamespace, rhpamKogitoImageStreamName, config.GetOperatorImageTag())
```
Comment (spolti) [inline test/pkg/installers/rhpam_kogito_installer.go:110]: shouldn't it be configurable?

Comment (sutaakar) [reply test/pkg/installers/rhpam_kogito_installer.go:110]: Do you mean `image-registry.openshift-image-registry.svc:5000`? That is OCP internal registry URL, it should be same everywhere.

**Review by spolti — APPROVED** [review]

**Comment by sutaakar** [comment]
I have noticed that sometimes there is an issue with image pull secret binding to the operator pod.
Applied the fix provided for upstream operator in https://github.com/kiegroup/kogito-operator/pull/327

**Review by radtriste — APPROVED** [review]

**Pull request merged**